### PR TITLE
Migrate sentinel-1 from etl

### PR DESCRIPTION
**Pull request by pjhartzell**
## Description

Adds the `sentinel-1-grd` and `sentinel-1-rtc` collections.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- Ingested Items into PC Test.
- Checked for equivalence to production Items by diffing an Item from the production API against one produced with this PR.

## Notes:

- Ran isort and black against the Python files.
- Docker images built and pushed to both `pccomponents` and `pccomponentstest` registries.

## Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review
- [ ] Changelog has been updated
- [ ] Documentation has been updated
- [ ] Unit tests pass locally (./scripts/test)
- [x] Code is linted and styled (./scripts/format)

### CODE REVIEW

File: datasets/sentinel-1-rtc/s1rtc.py [hunk 1]
```diff
@@ -57,11 +55,12 @@
 }
 
 
-class S1RTCCollection(Collection):
+class S1RTCCollection(Collection):  # type: ignore
```
Comment (TomAugspurger) [inline datasets/sentinel-1-rtc/s1rtc.py:58]: Out of curiosity, why the need for this `type: ignore`?

File: datasets/sentinel-1-rtc/dataset.yaml [hunk 2]
```diff
@@ -17,11 +18,12 @@ collections:
     template: ${{ local.path(./collection) }}
     class: s1rtc:S1RTCCollection
     asset_storage:
-      - uri: blob://sentinel1euwestrtc/sentinel1-grd-rtc-stac
+      - uri: blob://sentinel1euwestrtc/sentinel1-grd-rtc-stac/GRD/2023/5/  # testing
```
Comment (TomAugspurger) [inline datasets/sentinel-1-rtc/dataset.yaml:21]: LMK if you're able to handle setting these to values appropriate for running daily.

For goes-glm, we do have an optimization to only list assets under the year `2023`. We can do that with an `arg` in the `dataset.yaml` and have the caller pass in `--arg extra-prefix "2023/".

File: datasets/sentinel-1-rtc/s1rtc.py (hunk 1 above)
Comment (pjhartzell) [reply datasets/sentinel-1-rtc/s1rtc.py:58]: I'm not sure. I was getting a typing error, but no longer see it. I'll clean those up.

**Review by TomAugspurger — APPROVED** [review]
Thanks!

**Pull request merged**